hal/src/nRF52840/ble_hal.cpp

```diff
@@ -1458,7 +1462,7 @@ int BleObject::Observer::stopScanning() {
 
 ble_gap_scan_params_t BleObject::Observer::toPlatformScanParams() const {
     ble_gap_scan_params_t params = {};
-    params.extended = (scanParams_.scan_phys == BLE_PHYS_1MBPS) ? 0x00 : 0x01; /**< Extended required if other than PHYS_1MBPS */
+    params.extended = ( (scanParams_.scan_phys & BLE_PHYS_CODED) != 0) ? 0x01 : 0x00; /**< Extended must be 1 if using Coded PHY */
```

Comment:
I'd like to add a `uint8_t reserved[3]` filed in `hal_ble_scan_params_t` to make sure the struct is word aligned.

Reply:
Done